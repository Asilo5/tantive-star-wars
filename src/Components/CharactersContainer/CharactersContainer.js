import React from 'react';
import './CharactersContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard';
import NavBar from '../NavBar/NavBar';

const CharactersContainer = ({ characters }) => {
    const allCharacters = characters.map((character) => {
        return <CharacterCard key={character.name} {...character} />
    })
   return (
       <section className='characters-section'>
           <NavBar />
           <section className='characters'>
            {allCharacters} 
           </section>
       </section>
   )
}

export default CharactersContainer;