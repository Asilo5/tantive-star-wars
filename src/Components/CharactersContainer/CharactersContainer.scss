.characters {
    display: flex;
    flex-wrap: wrap;
    height: 61em;
    justify-content: center;
    left: 18%;
    overflow: scroll;
    padding: 20px;
    position: absolute;
    top: 44%;
 }