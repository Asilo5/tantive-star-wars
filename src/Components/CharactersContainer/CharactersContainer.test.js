import React from 'react';
import { shallow } from 'enzyme';
import CharactersContainer from './CharactersContainer';

describe('CharactersContainer', () => {

    const mockCharacters = [
      {name: "Luke Skywalker", homeworld: "Tatooine", population: "200000", species: "Human", films: Array(5)},
      {name: "C-3PO", homeworld: "Tatooine", population: "200000", species: "Droid", films: Array(6)},
      {name: "R2-D2", homeworld: "Naboo", population: "4500000000", species: "Droid", films: Array(7)},
      {name: "Darth Vader", homeworld: "Tatooine", population: "200000", species: "Human", films: Array(4)},
      {name: "Leia Organa", homeworld: "Alderaan", population: "2000000000", species: "Human", films: Array(5)},
      {name: "Obi-Wan Kenobi", homeworld: "Stewjon", population: "unknown", species: "Human", films: Array(6)},
      {name: "Chewbacca", homeworld: "Kashyyyk", population: "45000000", species: "Wookiee", films: Array(5)},
      {name: "Han Solo", homeworld: "Corellia", population: "3000000000", species: "Human", films: Array(4)},
      {name: "Jabba Desilijic Tiure", homeworld: "Nal Hutta", population: "7000000000", species: "Hutt", films: Array(3)},
      {name: "Wedge Antilles", homeworld: "Corellia", population: "3000000000", species: "Human", films: Array(3)}
    ]

    const wrapper = shallow(<CharactersContainer characters={mockCharacters} />)

    it('should match snapshot with all data passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})