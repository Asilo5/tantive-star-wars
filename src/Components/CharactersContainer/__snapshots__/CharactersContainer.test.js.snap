// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersContainer should match snapshot with all data passing in correctly 1`] = `
<section>
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Tatooine"
    name="Luke Skywalker"
    population="200000"
    species="Human"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Tatooine"
    name="C-3PO"
    population="200000"
    species="Droid"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Naboo"
    name="R2-D2"
    population="4500000000"
    species="Droid"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Tatooine"
    name="Darth Vader"
    population="200000"
    species="Human"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Alderaan"
    name="Leia Organa"
    population="2000000000"
    species="Human"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Stewjon"
    name="Obi-Wan Kenobi"
    population="unknown"
    species="Human"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Kashyyyk"
    name="Chewbacca"
    population="45000000"
    species="Wookiee"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Corellia"
    name="Han Solo"
    population="3000000000"
    species="Human"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Nal Hutta"
    name="Jabba Desilijic Tiure"
    population="7000000000"
    species="Hutt"
  />
  <[object Object]
    films={
      Array [
        undefined,
        undefined,
        undefined,
      ]
    }
    homeworld="Corellia"
    name="Wedge Antilles"
    population="3000000000"
    species="Human"
  />
</section>
`;
